app/decorators/models/solidus_address_name/address_decorator.rb

```diff
@@ -8,24 +8,18 @@ def self.prepended(base)
       base.alias_attribute :full_name, :name
 
       base.ignored_columns = []
+
       if base.method_defined?(:allowed_ransackable_attributes)
         base.allowed_ransackable_attributes.push(:firstname, :lastname)
       else
         base.whitelisted_ransackable_attributes.push(:firstname, :lastname)
       end
-    end
 
-    def name
-      "#{firstname} #{lastname}".strip
+      base.before_save :copy_split_name
```

Comment:
I am not sure about this callback. Do we want to do it for each time a record is saved?

Reply:
And can we have a spec for this functionality?